components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -941,9 +937,6 @@ private void updateOperationManagedInfo(SwaggerData.Resource resource, Operation
         // handle x-throttling-tier extension
         if (resource.getPolicy() != null && !StringUtils.isEmpty(resource.getPolicy())) {
             operation.setVendorExtension(APIConstants.SWAGGER_X_THROTTLING_TIER, resource.getPolicy());
-        } else if (resource.getThrottlingLimit() != null) {
-            operation.setVendorExtension(APIConstants.SWAGGER_X_THROTTLING_TIER,
```

Comment:
Is there a place to assign throttling-tier value from throttling-limit after removing this condition ? (same applies for the OAS3Parser as well.)

Reply:
if throttling limit is available, it means that policy wont be null. check URITemplate.java. And from there, the swaggerData.Resource is updated.

Reply:
Got it. 👍